File: j.cpp
#include <bits/stdc++.h>
using namespace std;

void lis(vector<int> &nums, int N){
	int ans = -1;
	if(N > 0){
		for(int i = 0 ; i < N ; i++){
			if(fabs(nums[N-1], nums[i]) <= 5)
				ans = max(lis(nums, N-1)+1)
		}
	}
}


int main(){
	int n, u, i;
	vector<int> nums;
	cin >> n;
	for(int i = 0 ; i < n ; i++){
		cin >> u;
		nums.push_back(u);
	}
	return 0;
}

File: j.py
from sys import stdin

read = stdin.readline

def main():
	n = int(read())
	nums = list(map(int, read().strip().split()))
	nums.sort()
	ans, ac, smaller = 1, 1, nums[0]
	for i in range(1, n):
		if nums[i]-smaller <= 5: ac += 1
		else: ac, ans, smaller = 1, max(ans, ac), nums[i]
	ans = max(ans, ac)
	print(ans)

main()